event-sourcing/event-repository/event-repository-core/src/main/java/uk/gov/justice/services/eventsourcing/repository/core/EventRepository.java

```diff
@@ -59,4 +59,9 @@
      */
     Long getCurrentSequenceIdForStream(final UUID streamId);
 
+    /**
+     * Returns stream of envelope streams. Envelopes in the nested stream are ordered by sequenceId
+     * @return the stream of envelope streams
+     */
+    Stream<Stream<JsonEnvelope>> getStreamOfEventStreams();
```

Comment:
Is it slightly clearer to have 'All' in the naming somewhere, to show that it is retrieving all events in stream of stream format.